Avoid unnecessary IF checks when generate Duration & Landing Time views

XD-DENG opened the pull request:
The original code is looping in a space which could be smaller, meanwhile IF check is not necessary.

This change aims for:
- bring MINOR performance improvement
- cleaner code

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

kaxil reviewed (commented):
I have rebased your PR on latest Master after applying Black and PyUpgrade

XD-DENG committed:
Avoid unnecessary IF checks when generate Duration & Landing Time views

The original code is looping in a space which could be smaller, meanwhile IF checks is not necessary.

This change aims for:
- bring MINOR performance improvement
- cleaner code
